sdk/servicebus/service-bus/src/core/streamingReceiver.ts

```diff
@@ -503,10 +528,14 @@ export class StreamingReceiver extends MessageReceiver {
         connectionId: this._context.connectionId,
         onError: (args) => this._onError && this._onError(args)
       });
-
-      this._receiverHelper.addCredit(this.maxConcurrentCalls);
+      const emptySlots = numberOfEmptyIncomingSlots(this.link);
+      const creditsToAdd =
```

Comment:
This same logic is in two spots - we should factor it out.

With that said, reading @ramya-rao-a 's comment above makes me think we need to add in testing for this. Please consider if you can do this in a unit-testable way, rather than running live testing and _if_ we decide we're not going to fix this in this PR leave a test that should pass when it does work and comment it out for now.

Reply:
Refactoring is a good idea, I'll work on it.
I'm unsure of adding more tests, I can add tests for the refactored methods though.

Reply:
Refactored